cluster/addons/dns/coredns/coredns.yaml.base

```diff
@@ -106,6 +106,8 @@ spec:
           effect: NoSchedule
         - key: "CriticalAddonsOnly"
           operator: "Exists"
+      nodeSelector:
+        beta.kubernetes.io/os: linux
```

Comment:
Is this label guaranteed to be present on node? When was this introduced?

Reply:
Just to sanity check myself out of pure paranoia, the value of this default label comes from goruntime.GOOS.

Gere are the permissible values as I understand them - 
[https://github.com/golang/go/blob/master/src/go/build/syslist.go]
const goosList = "aix android darwin dragonfly freebsd js linux nacl netbsd openbsd plan9 solaris windows zos "

I believe "linux" is the only value pertinent to coredns, given that the official Dockerfile is bound to debian.
[https://github.com/coredns/coredns/blob/master/Dockerfile]